docs/fluttertodostutorial.md

```diff
@@ -1,483 +1,1488 @@
+
 # Flutter Todos Tutorial
 
 ![advanced](https://img.shields.io/badge/level-advanced-red.svg)
 
 > In the following tutorial, we're going to build a Todos App in Flutter using the Bloc library.
 
 ![demo](./assets/gifs/flutter_todos.gif)
+> ZZTODO: Update gif with the movie in the issue tracker.
 
 ## Key Topics
 
 - Observe state changes with [BlocObserver](/coreconcepts?id=blocobserver).
 - [BlocProvider](/flutterbloccoreconcepts?id=blocprovider), Flutter widget which provides a bloc to its children.
 - [BlocBuilder](/flutterbloccoreconcepts?id=blocbuilder), Flutter widget that handles building the widget in response to new states.
-- Using Bloc instead of Cubit. [What's the difference?](/coreconcepts?id=cubit-vs-bloc)
+- [BlocListener](/flutterbloccoreconcepts?id=bloclistener), a Flutter widget which invokes the listener code in response to state changes in the bloc.
+- Using 3 Blocs and  1 Cubit. [What's the difference?](/coreconcepts?id=cubit-vs-bloc)
+- Uses a [Repository and API layer](architecture/repository) separately for the data.
+- [RespositoryProvider](/flutterbloccoreconcepts?id=respositoryprovider), a flutter widget to provide a repository to its children.
 - Prevent unnecessary rebuilds with [Equatable](/faqs?id=when-to-use-equatable).
-- [MultiBlocProvider](/flutterbloccoreconcepts?id=multiblocprovider), a Flutter widget that merges multiple BlocProvider widgets into one.
+- [MultiBlocListener](/flutterbloccoreconcepts?id=multibloclistener), a Flutter widget that merges multiple BlocListener widgets into one.
+
 
 ## Setup
 
 We'll start off by creating a brand new Flutter project
 
-[script](_snippets/flutter_todos_tutorial/flutter_create.sh.md ':include')
+```shell
+flutter create flutter_login
+```
 
 We can then replace the contents of `pubspec.yaml` with
 
 [pubspec.yaml](_snippets/flutter_todos_tutorial/pubspec.yaml.md ':include')
 
 and then install all of the dependencies
 
-[script](_snippets/flutter_todos_tutorial/flutter_packages_get.sh.md ':include')
+```shell
```

Comment:
Can we revert this change as well?

Reply:
Yes. This might be faster. I recommend making a list of final polish/styleguide steps here:

https://github.com/felangel/bloc/pull/3047#issuecomment-989334851

If you focus on formatting you'll never get through the tutorial.

It's faster to just write a list of style-guide items to implement (like a human-implemented css file) than it is to make the individual changes.

Take a look via docsify. There are notes to ignore in the raw markdown.